### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by Srykah** [OpMonTeam/OpMon-Archive#116]
The documentation is lacking or outdated in a lot of areas ; a lot of spelling errors and bad translations are also present. I open this issue to break the problem in smaller tasks so we can see how we progress (plus ticking checkmarks is quite enjoyable don't you think ? ^^)

Some items are already done and up-to-date, and I have probably forgotten a bunch of things too. Please correct me below (and notify when you've documented anything so I can update the list !) For items with too many sub-items, an asterisk (*) will be present.

Also, from now on, contributions should always be accompanied with their relative documentation, so as to avoid the state we're currently in.

- [ ] Code
  - [ ] Global code structure
    - [ ] Model-View-Controller paradigm and link between related classes
    - [ ] Internal and external dependencies, how and why they're used
  - [x] Individual files (document everything in the headers with Doxygen) (*)
  - [ ] Build environment
    - [ ] CMake
    - [ ] Clang-format & clang-tidy
    - [x] Folder structure
- [ ] Project organization
  - [x] How to contribute
    - [x] Code
    - [x] Art
    - [x] OpMons
    - [ ] Translations
    - [x] Documentation
  - [ ] List of all the project's resources and how they are tied together (other repos, the discord, the wiki, the website etc.)
- [ ] The game
  - [ ] How to play
    - [x] Install and launch the game
    - [ ] Controls
    - [ ] Troubleshooting
  - [x] OpMons (*)
  - [ ] Strategy guide
    - [ ] Types and type effectiveness (single page)
    - [ ] Attacks (*)
    - [ ] Items (*)
    - [ ] Catch and train OpMons
  - [ ] Story
    - [ ] The plot (per chapter) (*)
    - [ ] Characters (*)
    - [ ] Types of trainers and their OpMons (and unique, story-related trainers) (*)
  - [ ] The world
    - [ ] Geography
      - [ ] Biomes (*) 
      - [ ] Cities (*)
    - [ ] Politics
    - [ ] History
  - [ ] Data representation (aka JSON)
    - [x] OpMons
    - [x] Attacks
    - [ ] Items
    - [x] Maps
    - [ ] Cutscenes
    - [x] Trainers
    - [ ] Settings
    - [ ] Localization strings
    - [x] Issue #138

**Comment by milkyroute** [comment]
I totally agree, thank you for having took the time to summarize everything. I think this is what I will focus on for now, because it's a big problem since the very beginning of the development.
